File: telerik_ui_for_winforms_courseware_chm/Courseware/Scheduler/VB/BusinessObjects/BusinessObjects/MyAppointment.vb
Imports System.ComponentModel

Public Class MyAppointment
    Implements INotifyPropertyChanged
    Private start_Renamed As DateTime = DateTime.Now
    Private end_renamed As DateTime = DateTime.Now
    Private subject_Renamed As String = String.Empty
    Private description_Renamed As String = String.Empty
    Private location_Renamed As String = String.Empty
    Private id_Renamed As Guid = Guid.NewGuid()

    Public Sub New()
    End Sub

    Public Sub New(ByVal start_Renamed As DateTime, ByVal end_renamed As DateTime, ByVal subject_Renamed As String, ByVal description_Renamed As String, ByVal location_Renamed As String)
        Me.start_Renamed = start_Renamed
        Me.end_renamed = end_renamed
        Me.subject_Renamed = subject_Renamed
        Me.description_Renamed = description_Renamed
        Me.location_Renamed = location_Renamed
    End Sub

    Public Property Id() As Guid
        Get
            Return Me.id_Renamed
        End Get
        Set(ByVal value As Guid)
            If Me.id_Renamed <> value Then
                Me.id_Renamed = value
                Me.OnPropertyChanged("Id")
            End If
        End Set
    End Property

    Public Property Start() As DateTime
        Get
            Return Me.start_Renamed
        End Get
        Set(ByVal value As DateTime)
            If Me.start_Renamed <> value Then
                Me.start_Renamed = value
                Me.OnPropertyChanged("Start")
            End If
        End Set
    End Property

    Public Property [End]() As DateTime
        Get
            Return Me.end_renamed
        End Get
        Set(ByVal value As DateTime)
            If Me.end_renamed <> value Then
                Me.end_renamed = value
                Me.OnPropertyChanged("End")
            End If
        End Set
    End Property

    Public Property Subject() As String
        Get
            Return Me.subject_Renamed
        End Get
        Set(ByVal value As String)
            If Me.subject_Renamed <> value Then
                Me.subject_Renamed = value
                Me.OnPropertyChanged("Subject")
            End If
        End Set
    End Property

    Public Property Description() As String
        Get
            Return Me.description_Renamed
        End Get
        Set(ByVal value As String)
            If Me.description_Renamed <> value Then
                Me.description_Renamed = value
                Me.OnPropertyChanged("Description")
            End If
        End Set
    End Property

    Public Property Location() As String
        Get
            Return Me.location_Renamed
        End Get
        Set(ByVal value As String)
            If Me.location_Renamed <> value Then
                Me.location_Renamed = value
                Me.OnPropertyChanged("Location")
            End If
        End Set
    End Property

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Protected Overridable Sub OnPropertyChanged(ByVal propertyName As String)
        If Not Me.PropertyChangedEvent Is Nothing Then
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End If
    End Sub
End Class

File: telerik_ui_for_winforms_courseware_chm/Courseware/Scheduler/CS/BusinessObjects/BusinessObjects/MyAppointment.cs
using System;
using System.ComponentModel;

namespace BusinessObjects
{
    public class MyAppointment : INotifyPropertyChanged
    {
        private DateTime start = DateTime.Now;
        private DateTime end = DateTime.Now;
        private string subject = string.Empty;
        private string description = string.Empty;
        private string location = string.Empty;
        private Guid id = Guid.NewGuid();

        public MyAppointment()
        {
        }

        public MyAppointment(DateTime start, DateTime end, string subject, string description, string location)
        {
            this.start = start;
            this.end = end;
            this.subject = subject;
            this.description = description;
            this.location = location;
        }

        public Guid Id
        {
            get
            {
                return this.id;
            }
            set
            {
                if (this.id != value)
                {
                    this.id = value;
                    this.OnPropertyChanged("Id");
                }
            }
        }

        public DateTime Start
        {
            get
            {
                return this.start;
            }
            set
            {
                if (this.start != value)
                {
                    this.start = value;
                    this.OnPropertyChanged("Start");
                }
            }
        }

        public DateTime End
        {
            get
            {
                return this.end;
            }
            set
            {
                if (this.end != value)
                {
                    this.end = value;
                    this.OnPropertyChanged("End");
                }
            }
        }

        public string Subject
        {
            get
            {
                return this.subject;
            }
            set
            {
                if (this.subject != value)
                {
                    this.subject = value;
                    this.OnPropertyChanged("Subject");
                }
            }
        }

        public string Description
        {
            get
            {
                return this.description;
            }
            set
            {
                if (this.description != value)
                {
                    this.description = value;
                    this.OnPropertyChanged("Description");
                }
            }
        }

        public string Location
        {
            get
            {
                return this.location;
            }
            set
            {
                if (this.location != value)
                {
                    this.location = value;
                    this.OnPropertyChanged("Location");
                }
            }
        }

        #region INotifyPropertyChanged Members

        public event PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged(string propertyName)
        {
            if (this.PropertyChanged != null)
            {
                this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

        #endregion
    }
}
